pep/unary_client.go

```diff
@@ -84,7 +84,16 @@ func (c *unaryClient) Connect(addr string) error {
 		return err
 	}
 
-	conn, err := grpc.Dial(addr, opts...)
+	var conn *grpc.ClientConn
+	if c.opts.connTimeout > 0 {
+		opts = append(opts, grpc.WithBlock())
+		toCtx, toCancelFn := context.WithTimeout(context.Background(), c.opts.connTimeout)
+		defer toCancelFn()
+		conn, err = grpc.DialContext(toCtx, addr, opts...)
```

Comment:
Note that with the timeout, Connect() becomes a blocking call.  Without the timeout, Connect() remains a non-blocking call.